### ISSUE TITLE
sl web does not work well on Windows

### ISSUE THREAD

**Issue by AngelEzquerra** [facebook/sapling#173]
On windows, the sl web command does open a web page but shows the following error:

```
Failed to fetch Diffs
Command failed with exit code 1: gh api graphql -f searchQuery=repo:AngelEzquerra/Arraymancer is:pr author:@me -F numToFetch=100 -f query= query YourPullRequestsQuery($searchQuery: String!, $numToFetch: Int!) { search(query: $searchQuery, type: ISSUE, first: $numToFetch) { nodes { ... on PullRequest { __typename number title state url comments { totalCount } commits(last: 1) { nodes { commit { checkSuites(first: 100) { nodes { conclusion status } } } } } } } } } 'gh' is not recognized as an internal or external command, operable program or batch file.
Error: Command failed with exit code 1: gh api graphql -f searchQuery=repo:AngelEzquerra/Arraymancer is:pr author:@me -F numToFetch=100 -f query=
    query YourPullRequestsQuery($searchQuery: String!, $numToFetch: Int!) {
  search(query: $searchQuery, type: ISSUE, first: $numToFetch) {
    nodes {
      ... on PullRequest {
        __typename
        number
        title
        state
        url
        comments {
          totalCount
        }
        commits(last: 1) {
          nodes {
            commit {
              checkSuites(first: 100) {
                nodes {
                  conclusion
                  status
                }
              }
            }
          }
        }
      }
    }
  }
}
    
'gh' is not recognized as an internal or external command,
operable program or batch file.
    at e.exports (C:\Program Files\Sapling\edenscm-isl\isl-server\dist\387.js:1:18531)
    at C:\Program Files\Sapling\edenscm-isl\isl-server\dist\387.js:1:15713
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

I'm assuming that gh is the github CLI tool which is installed and on the path.
Another problem is that I am running `sl web` on a repo called "poc" but the error message references a different repo (Arraymancer).

**Comment by AngelEzquerra** [comment]
Thank you Evan, that did help. I get a list of files now.
However I still get an error:

```
Log in to gh CLI with `gh auth login` to allow requests to GitHub
```

This is on a non github based repo so it seems that it always tries to access github, even in repos that were not cloned from github?

**Comment by evangrayk** [comment]
Ah, yes we are kind of assuming right now in the UI that any git based repo is backed by github, but we don't intend for that to be a permanent assumption. 

Perhaps a better strategy is for us to show this error once, and if you dismiss it, turn "off" pull request integrations like submitting or limit it to some kind of basic `sl push` command. Another idea would be to detect github specifically, though I'm not sure how that will play with things like self-hosted github servers.

**Comment by evangrayk** [comment]
Just wondering, are you using github enterprise or a separate provider entirely like gitlab?

**Comment by bolinfest** [comment]
I think many of these things should be fixed in `sl web` in this morning's release!

**Comment by AngelEzquerra** [comment]
@evangrayk, I use regular GitHub in some repos, and a private bitbucket server in others (including the one in which there was this failure).

**Comment by AngelEzquerra** [comment]
I tried the new version and now the main page does not load. I just get the following error:

```
Something went wrongCannot read properties of undefined (reading 'message')TypeError: Cannot read properties of undefined (reading 'message') at d (http://localhost:3011/static/js/main.js:2:3931) at xa (http://localhost:3011/static/js/main.js:2:314856) at Cl (http://localhost:3011/static/js/main.js:2:372096) at bu (http://localhost:3011/static/js/main.js:2:361032) at gu (http://localhost:3011/static/js/main.js:2:360960) at mu (http://localhost:3011/static/js/main.js:2:360823) at ou (http://localhost:3011/static/js/main.js:2:357603) at lu (http://localhost:3011/static/js/main.js:2:357994) at Bi (http://localhost:3011/static/js/main.js:2:298577) at http://localhost:3011/static/js/main.js:2:355536
```

**Comment by evangrayk** [comment]
@AngelEzquerra  thanks for trying the new version. That error looks like it might be from a version mismatch between the client and server. Could you try using `sl web --force` to start a new server?

There should be a fix in the new version that forces new servers if all smartlog windows are closed: 8b1a3c8fe66e, which should prevent that in the future. I'm also planning on adding some version number checking to show a better error message if versions get out of sync or the server is stale.
